compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIECreateLogicalObjectFifoLink.cpp

```diff
@@ -13,6 +14,49 @@
 
 namespace mlir::iree_compiler::AMDAIE {
 
+/// Utility to verify that there are no overlapping access patterns between
```

Comment:
> Just to check: you're not interested in whether the patterns actually access the same elements?

Correct, this would still be considered an overlap as they access the same region of the buffer and this is currently not handled by the backend.

> Assuming that this has nothing to do with interleaving/non-interleaving, what is the problem specifically that MLIR-AIE has? I'm coming at this with a very shallow understanding of mlir-aie op semantics..

It can be supported in HW, even when consumers/producers are accessing the same elements, however, the aie.objectfifo ops don't have semantics for this at the moment as the order and sizes are used to implicitly split a buffer access across multiple producer/consumer actors, see https://github.com/Xilinx/mlir-aie/blob/061bc4a01d97f58ff0f2d63ecef099ca9c41b8ee/include/aie/Dialect/AIE/IR/AIEOps.td#L1685. And this works, because in most use cases, you won't see this overlap happening, but we might want to support it at some point (and there is some work on supporting offsets in the link operation in MLIR-AIE).